misc-utils/lsfd.c

```diff
@@ -496,11 +496,11 @@ static const struct file_class *stat2class(struct stat *sb)
 		return &file_class;
 	case S_IFREG:
 		dev = sb->st_dev;
-		if (major(dev) != 0)
+		if (major(dev) != 0 || sb->st_size != 0)
 			return &file_class;
 
 		fs = get_nodev_filesystem(minor(dev));
-		if (fs && strcmp(fs, "nsfs") == 0)
+		if (!fs || strcmp(fs, "nsfs") == 0)
```

Comment:
The condition `!fs` looks too relaxed. I wonder we can use is_nsfs_dev() like:
```
diff --git a/misc-utils/lsfd.c b/misc-utils/lsfd.c
index 251d1dc93..0b7e4aa96 100644
--- a/misc-utils/lsfd.c
+++ b/misc-utils/lsfd.c
@@ -501,8 +501,10 @@ static const struct file_class *stat2class(struct stat *sb)
 		if (major(dev) != 0)
 			return &file_class;
 
-		fs = get_nodev_filesystem(minor(dev));
-		if (fs && strcmp(fs, "nsfs") == 0)
+		if (is_nsfs_dev(sb.st_dev))
+			return &nsfs_file_class;
+		else if ((fs = get_nodev_filesystem(minor(dev)))
+			 && strcmp(fs, "nsfs") == 0)
 			return &nsfs_file_class;
 
 		return &file_class;
```

I will take more time for this.

Reply:
It can be so easy :-)
I'll integrate this.